packages/dds/tree/docs/wip/Modules.md

```diff
@@ -0,0 +1,196 @@
+# Modules
+
+There are a wide variety of use-cases for a Fluid tree, and there are several trade-offs in the design space that should be made differently for the different use-cases.
+To enable flexibility both for how our users configure the system, as well as when we modify the implementation,
+the implementation consist of collection of modules which can be composed in a few different ways to address a wide variety of use cases.
+
+This breakdown is intended as an architectural approach to ensure we can deliver functionality in an incremental way (minimizing the work needed for the early milestones),
+as well as quickly respond to specific feature requests, and perform targeted unit testing.
+
+These module boundaries could additionally be exposes as public packages,
+allowing our consumers to also extend and configure the functionality in these ways (increasing our internal API stability obligations),
+however that is a separate decision which is considered out of scope here.
```

Comment:
Okay.  I'm guessing that by "module", you mean a subdirectory of related things that are explicitly exported by an index.ts:

tree/schema/index.ts
tree/forest/index.ts
tree/rebaser/index.ts
...

and so on...

Reply:
Module here intended to use the JavaScript / TypeScript meaning (aka: a file you can import/requite as defined in https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Modules). In our codebase that takes the form of a typescript file, which might be an index.ts file representing a directory, or some other typescript file.